### ISSUE TITLE
False positive for javascript:S2392

### ISSUE THREAD

**Issue by alixwar** [SonarSource/sonar-javascript#753]
**I want to report a bug.**

**SonarJS version:** 3.1.1.5128<br>
**SonarQube version:** 5.6.6<br>

**Rule key:** javascript:S2392

**Reproducer** 
```
// False-positive for javascript:S2392
let a, b;

// This is OK
// let a = null;
// let b = null;


for (let i = 0; i < 3; i++) {
  if (i === 1) {
     a = true;
  }
  if (i === 2) {
     b = true;
  }
  if (a && b) {
    console.debug('same');
  }
}
```

**I want to provide the following feedback.**
Following the recommendation by the rule description and move a and b declarations into the loop would result in changed behavior as both values would then be recomputed every iteration. Therefore I thinks it's a false positive and not a false negative.

Regards Alix

**Comment by vilchik-elena** [comment]
Hi,
thanks for feedback! Indeed it's a false positive, there is an issue for it already, #712
I will close this as a duplicate
